src/main/java/com/github/harti2006/neo4j/StartNeo4jServerMojo.java

```diff
@@ -120,8 +151,49 @@ private void startNeo4jServer() throws MojoExecutionException {
             } else {
                 throw new MojoExecutionException("Neo4j server did not start up properly");
             }
+            
+            // Now we need to wait it replies
+            checkServerReady ();
+            
+            // Went up! If it's a new DB, let's change the password.
+            setNewPassword ();
         } catch (IOException | InterruptedException e) {
-            throw new MojoExecutionException("Could not start neo4j server", e);
+            throw new MojoExecutionException( "Could not start neo4j server", e);
         }
     }
+    
+    private void checkServerReady () throws MojoExecutionException, InterruptedException
+    {
+    		String pwd = deleteDb ? "neo4j" : password;
+    		int maxAttempts = 3;
+    		Thread.sleep ( 1500 ); // It takes some time anyway
+    		
+      for ( int attempts = 1; attempts <= maxAttempts ; attempts++ )
+      {
+      		try 
+      		{
+      			getLog().debug ( "Trying to connect Neo4j, attempt " + attempts ); 
+	        Driver driver = GraphDatabase.driver( "bolt://127.0.0.1:" + boltPort, AuthTokens.basic ( "neo4j", pwd ) );
+	      		driver.close ();
+	      		return;
+      		}
+      		catch ( ServiceUnavailableException ex ) {
+      		}
+      		Thread.sleep ( 3000 );
+      }
+      throw new MojoExecutionException ( 
+      		"Server doesn't result started after waiting for its boot" 
+      	);
+    }
+    
+    private void setNewPassword ()
+    {
+      if ( !deleteDb ) return;
+      
+      Driver driver = GraphDatabase.driver( "bolt://127.0.0.1:" + boltPort, AuthTokens.basic ( "neo4j", "neo4j" ) );
```

Comment:
Would there be any benefit in re-using the driver that has been created during `checkServerReady`?